### DISCUSSION TITLE
task 01 - Q1

### DISCUSSION THREAD

**Discussion by SMRazaRizvi96** [chulminy/CIVE497-CIVE700#149] [Q&A] [answered]
While visualizing the cameraman.tif, the `imshow()` function appears to be using the wrong colormap, resulting in the following output instead of a grayscale image:

![Screenshot from 2024-01-12 22-45-54](https://github.com/chulminy/CIVE497-CIVE700/assets/74411560/8a285a5b-b4fc-4c44-9519-9f254af0dfeb)

- Is it the same for everyone?

- Should I fix this in the code or leave it as is, as the question asks us to just use the same code with a different image file?

**Comment by chulminy** [comment, marked as answer]
Raza

I think it's the issue of this:

![image](https://github.com/chulminy/CIVE497-CIVE700/assets/3227103/2a7ee7b7-c736-4bac-82bd-898edf5a075b)

Chul Min

**Reply by SMRazaRizvi96** [reply]
Oh alright. I'll make the changes to my code to use cv2 instead of PIL.

I thought we just had to run the same code as the question said _"Please run this code with 'cameraman.tif' in the 'img' folder"_.

**Reply by noreengao0924** [reply]
I'm still confused because the example code in Q1 does not use cv2. Is it okay if I change the cmap to 'gray' when I use imshow to be able to show the image in grayscale?

**Reply by chulminy** [reply]
You can use cv2 or PIL. That's not a problem of cv2. When the cv2 read the image, the order is BGR, not RGB. So, when you use cv2, you need to use the script highlighted above. Then, nothing different.
